modules/apps/27-interchain-accounts/host/keeper/msg_server.go

```diff
@@ -23,6 +26,40 @@ func NewMsgServerImpl(keeper *Keeper) types.MsgServer {
 	return &msgServer{Keeper: keeper}
 }
 
+// ModuleQuerySafe routes the queries to the keeper's query router if they are module_query_safe.
+// This handler doesn't use the signer.
+func (m msgServer) ModuleQuerySafe(goCtx context.Context, msg *types.MsgModuleQuerySafe) (*types.MsgModuleQuerySafeResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	responses := make([][]byte, len(msg.Requests))
+	for i, query := range msg.Requests {
+		isModuleQuerySafe := slices.Contains(m.mqsWhitelist, query.Path)
+		if !isModuleQuerySafe {
+			return nil, errorsmod.Wrapf(ibcerrors.ErrInvalidRequest, "cannot verify module query safe: %s", query.Path)
```

Comment:
I don't know, I found the usage of "verify" a bit confusing...

```suggestion
			return nil, errorsmod.Wrapf(ibcerrors.ErrInvalidRequest, "not a module query safe: %s", query.Path)
```

Reply:
done